File: Fundamentals-of-Programing142-main/homework/objective_03/2pt_make_change/make_change.cpp
/*************************************************************************
 * Objective 4: A change computation program
 *
 * File Name: make_change.cpp
 * Name:      ?
 * Course:    CPTR 142
 */

#include <iostream>
#include <iterator>
#include <ostream>

int main() {
  // define variables
  int dollar_20, dollar_10, dollar_5, dollar_1, c_25, c_10, c_5, c_1, payback;
  double cost, paid;
  // Prompt for and collect input
  std::cout << "Enter the cost in dollars and cents: $";
  std::cin >> cost;
  std::cout << "Enter the amount paid in dollars and cents: $";
  std::cin >> paid;
  // convert to cents and compute change
  cost *= 100;
  paid *= 100;
  payback = paid - cost;
  dollar_20 = payback / 2000;
  payback %= 2000;
  dollar_10 = payback / 1000;
  payback %= 1000;
  dollar_5 = payback / 500;
  payback %= 500;
  dollar_1 = payback / 100;
  payback %= 100;
  c_25 = payback / 25;
  payback %= 25;
  c_10 = payback / 10;
  payback %= 10;
  c_5 = payback / 5;
  payback %= 5;
  c_1 = payback / 1;
  payback %= 1;
  // now display number of dollars and coins
  std::cout << std::endl << "Your change is:" << std::endl;
  std::cout << "  - twenty-dollar bills: " << dollar_20 << std::endl;
  std::cout << "  - ten-dollar bills: " << dollar_10 << std::endl;
  std::cout << "  - five-dollar bills: " << dollar_5 << std::endl;
  std::cout << "  - one-dollar bills: " << dollar_1 << std::endl;
  std::cout << "  - quarters: " << c_25 << std::endl;
  std::cout << "  - dimes: " << c_10 << std::endl;
  std::cout << "  - nickels: " << c_5 << std::endl;
  std::cout << "  - pennies: " << c_1 << std::endl;

  return 0;
}

File: Fundamentals-of-Programing141-main/homework/02_variables_expressions/04_expert/02_make_change/make_change.py
"""
Expert Problem: Make change using various currency

File Name: make_change.py
Name:      ?
Course:    CPTR 141
"""

# Your code goes here
cost = float(input("Enter the cost in dollars and cents: $"))
paid = float(input("Enter the amount paid in dollars and cents: $"))
reminder = paid - cost
bill20 = reminder//20
reminder -= bill20*20
bill10 = reminder//10
reminder -= bill10*10
bill5 = reminder//5
reminder -= bill5*5
bill1 = reminder//1
reminder -= bill1*1
quarters = reminder//0.25
reminder -= quarters*0.25
dimes = reminder//0.10
reminder -= dimes*0.10
nickels = reminder//0.05
reminder -= nickels*0.05
pennies = reminder//0.01

print("\nYour change is:")
print("  - twenty-dollar bills:", int(bill20))
print("  - ten-dollar bills:",int(bill10))
print("  - five-dollar bills:",int(bill5))
print("  - one-dollar bills:",int(bill1))
print("  - quarters:",int(quarters))
print("  - dimes:",int(dimes))
print("  - nickels:",int(nickels))
print("  - pennies:",int(pennies))